internal/repository/flight.go

```diff
@@ -0,0 +1,40 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/dilyara4949/flight-booking-api/internal/domain"
+	errs "github.com/dilyara4949/flight-booking-api/internal/repository/errors"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+	"time"
+)
+
+type FlightRepository struct {
+	db *gorm.DB
+}
+
+func NewFlightRepository(db *gorm.DB) FlightRepository {
+	return FlightRepository{db: db}
+}
+
+func (repo *FlightRepository) Get(ctx context.Context, id uuid.UUID, available bool) (*domain.Flight, error) {
+	flight := domain.Flight{}
+
+	query := repo.db.WithContext(ctx)
+
+	if available {
```

Comment:
Saw similar piece of code in another function, this might be a good candidate to have itself a sepoarate function =)